src/main/java/net/dv8tion/jda/api/JDABuilder.java

```diff
@@ -992,7 +1443,12 @@ public JDA build() throws LoginException
         MetaConfig metaConfig = new MetaConfig(maxBufferSize, contextMap, cacheFlags, flags);
 
         JDAImpl jda = new JDAImpl(authConfig, sessionConfig, threadingConfig, metaConfig);
-        jda.setChunkingFilter(chunkingFilter);
+        jda.setMemberCachePolicy(memberCachePolicy);
+        // We can only do member chunking with the GUILD_MEMBERS intent
+        if ((intents & GatewayIntent.GUILD_MEMBERS.getRawValue()) == 0)
+            jda.setChunkingFilter(ChunkingFilter.NONE);
```

Comment:
We can't do that because the chunking filter is a function. There is no way for us to check whether this function would be always true, always false, or anything in between. We can only check if its any of the configured defaults but I think its best to just make the choice for the user. This is documented in the `GatewayIntent` enum.

Reply:
What if the `chunkingFilter ` was left as `null` instead of populated with `ChunkingFilter.ALL`?
Then we could see if a user explicitly set a chunking filter (as it would be non-null). This would allow us to say "you cannot set a chunking filter when `GUILD_MEMBERS` is disabled.

If the value was still null however we would be free to make our own decision.
(`ALL` when GUILD_MEMBERS, `NONE` when not)

Reply:
I'd just log a warning if `chunkingFilter != NONE` and be done with it